autonomic-administration-plugin/src/test/java/br/com/autonomiccs/autonomic/administration/plugin/services/AutonomicClusterManagementServiceTest.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * This program is part of Autonomiccs "autonomic-platform",
+ * an open source autonomic cloud computing management platform.
+ * Copyright (C) 2016 Autonomiccs, Inc.
+ *
+ * Licensed to the Autonomiccs, Inc. under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The Autonomiccs, Inc. licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package br.com.autonomiccs.autonomic.administration.plugin.services;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InOrder;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import com.cloud.dc.ClusterVO;
+import com.cloud.dc.dao.ClusterDao;
+
+import br.com.autonomiccs.autonomic.administration.algorithms.ClusterAdministrationHeuristicAlgorithm;
+import br.com.autonomiccs.autonomic.plugin.common.daos.ClusterJdbcDao;
+import br.com.autonomiccs.autonomic.plugin.common.enums.ClusterAdministrationStatus;
+
+@RunWith(PowerMockRunner.class)
+public class AutonomicClusterManagementServiceTest {
+
+    private AutonomicClusterManagementService autonomicClusterManagementService;
+    @Mock
+    private ClusterJdbcDao clusterDaoJdbc;
+    @Mock
+    private ClusterDao clusterDao;
+
+    @Before
+    public void setup() {
+        autonomicClusterManagementService = Mockito.spy(new AutonomicClusterManagementService());
+        autonomicClusterManagementService.clusterDaoJdbc = clusterDaoJdbc;
+        autonomicClusterManagementService.clusterDao = clusterDao;
+    }
+
+    @Test
+    public void isClusterBeingAdministratedTest() {
```

Comment:
what is this test actually testing? Im not so familiar with Mocks and the like but to me it seems that this test is not functional. You are just testing if it returns an already set expression. Shouldn't you mock a call for administration and then test if the cluster's status is InProgress?